compiler/rustc_codegen_llvm/src/abi.rs

```diff
@@ -362,9 +362,14 @@ impl<'ll, 'tcx> FnAbiLlvmExt<'ll, 'tcx> for FnAbi<'tcx, Ty<'tcx>> {
                         // currently use this mode so we have to allow it -- but we absolutely
                         // shouldn't let any more targets do that.
                         // (Also see <https://github.com/rust-lang/rust/issues/115666>.)
+                        //
+                        // The unstable abi `PtxKernel` also uses Direct for now.
+                        // It needs to switch to something else before stabilization can happen.
+                        // (Tracking issue abi_ptx: https://github.com/rust-lang/rust/issues/38788)
```

Comment:
Can you open an issue specifically for "nvptx uses Direct for Aggregates", and link that from here and from #38788?